### PR TITLE
Release `v1.2.1`

### PR DESCRIPTION
**Pull request by sphuber**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Changelog
 
+
+## v1.2.1
+
+In the fixing of three bugs, three minor features have been added along the way.
```
Comment (ltalirz) [inline CHANGELOG.md:6]: do we now start to have taglines per release? ;-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+
+### Features
+- Add config option `daemon.worker_process_slots` to configure the maximum number of concurrent task each daemon worker can handle [[#3949]](https://github.com/aiidateam/aiida-core/pull/3949)
```
Comment (ltalirz) [inline CHANGELOG.md:9]:
```suggestion
- Add config option `daemon.worker_process_slots` to configure the maximum number of concurrent tasks each daemon worker can handle [[#3949]](https://github.com/aiidateam/aiida-core/pull/3949)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+- Add config option `daemon.default_workers` to set the default number of workers to be started by `verdi daemon start` [[#3949]](https://github.com/aiidateam/aiida-core/pull/3949)
+- `CalcJob`: make submit script filename configurable through the `metadata.options` [[#3948]](https://github.com/aiidateam/aiida-core/pull/3948)
+
+### Bug fixes
+- `CalcJob`: fix bug in idempotency check of upload transport task [[#3948]](https://github.com/aiidateam/aiida-core/pull/3948)
+- REST API: reintroduce CORS headers [[#3951]](https://github.com/aiidateam/aiida-core/pull/3951)
```
Comment (ltalirz) [inline CHANGELOG.md:15]:
```suggestion
- REST API: reintroduce CORS headers (was breaking Materials Cloud provenance explorer) [[#3951]](https://github.com/aiidateam/aiida-core/pull/3951)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+- Remove the equality operator of `ExitCode` which caused the serialization of workchains to fail if put in the workchain context [[#3940]](https://github.com/aiidateam/aiida-core/pull/3940)
+
+### Deprecations
+- The `--hookup` argument of `aiida.restapi.run_api` is deprecated [[#3951]](https://github.com/aiidateam/aiida-core/pull/3951)
```
Comment (ltalirz) [inline CHANGELOG.md:19]:
```suggestion
- The `hookup` argument of `aiida.restapi.run_api` and the `--hookup` option of `verdi restapi` are deprecated [[#3951]](https://github.com/aiidateam/aiida-core/pull/3951)
```

File: CHANGELOG.md (hunk 1 above)
Comment (sphuber) [reply CHANGELOG.md:6]: Just felt the need to justify there being some features in a patch release. You think I should remove it?

Comment (ltalirz) [reply CHANGELOG.md:6]: It's fine :-)

**Pull request merged**